### ISSUE TITLE
Found multiple publish output files with the same relative path: `D:\myproject\Presentation\Nop.Web\Views\_ViewImports.cshtml`, `D:\myproject\Plugins\Nop.Plugin.Shipping.ShipStation\Views\_ViewImports.cshtml`

### ISSUE THREAD

**Issue by Manuel-Innovapps** [nopSolutions/nopCommerce#6096]
nopCommerce version: 4.50.0 (full release from today)

Also happens in a completely clean installation

Steps to reproduce the problem: 
* Download the latest release and extract it to somewhere
* With the terminal cd in your directory
* run `dotnet build`
* run `dotnet publish -o ./myapp`

We initially saw this in our ci/cd pipeline. Our pipelines are failing. We cannot deploy at the moment.
I think that we have an early 4.5 beta somewhere, where we have a pipeline setup which works. So my guess is that the error was introduced after the initial nop 4.5 beta release.

Logs:
```shell
PS D:\myproject> dotnet publish -o ./myapp 
Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  Nop.Core -> D:\myproject\Libraries\Nop.Core\bin\Debug\net6.0\Nop.Core.dll
  Nop.Core -> D:\myproject\myapp\
  Nop.Data -> D:\myproject\Libraries\Nop.Data\bin\Debug\net6.0\Nop.Data.dll
  Nop.Data -> D:\myproject\myapp\
  Nop.Services -> D:\myproject\Libraries\Nop.Services\bin\Debug\net6.0\Nop.Services.dll
  Nop.Services -> D:\myproject\myapp\
  Nop.Web.Framework -> D:\myproject\Presentation\Nop.Web.Framework\bin\Debug\net6.0\Nop.Web.Framework.dll
  Nop.Plugin.ExchangeRate.EcbExchange -> D:\myproject\Presentation\Nop.Web\Plugins\ExchangeRate.EcbExchange\Nop.Plugin.ExchangeRate.EcbExchange.dll
  Nop.Web.Framework -> D:\myproject\myapp\
  Nop.Plugin.Widgets.AccessiBe -> D:\myproject\Presentation\Nop.Web\Plugins\Widgets.AccessiBe\Nop.Plugin.Widgets.AccessiBe.dll
  Nop.Plugin.ExchangeRate.EcbExchange -> D:\myproject\myapp\
  Nop.Plugin.Shipping.FixedByWeightByTotal -> D:\myproject\Presentation\Nop.Web\Plugins\Shipping.FixedByWeightByTotal\Nop.Plugin.Shipping.FixedByWeightByTotal.dll
  Nop.Plugin.Tax.FixedOrByCountryStateZip -> D:\myproject\Presentation\Nop.Web\Plugins\Tax.FixedOrByCountryStateZip\Nop.Plugin.Tax.FixedOrByCountryStateZip.dll
  Nop.Plugin.Pickup.PickupInStore -> D:\myproject\Presentation\Nop.Web\Plugins\Pickup.PickupInStore\Nop.Plugin.Pickup.PickupInStore.dll
D:\myproject\Plugins\Innovapps.Nop.Plugin.Misc.BusinessCentralErp\Areas\Admin\Controllers\BusinessCentralErpQueueListController.cs(86,55): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [D:\myproject\Plugins\Innovapps.Nop.Plugin.Misc.BusinessCentralErp\Innovapps.Nop.Plugin.Misc.BusinessCentralErp.csproj]
D:\myproject\Plugins\Innovapps.Nop.Plugin.Misc.BusinessCentralErp\Areas\Admin\Controllers\BusinessCentralErpQueueListController.cs(143,58): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [D:\myproject\Plugins\Innovapps.Nop.Plugin.Misc.BusinessCentralErp\Innovapps.Nop.Plugin.Misc.BusinessCentralErp.csproj]
  Nop.Plugin.Payments.CheckMoneyOrder -> D:\myproject\Presentation\Nop.Web\Plugins\Payments.CheckMoneyOrder\Nop.Plugin.Payments.CheckMoneyOrder.dll
  Nop.Plugin.MultiFactorAuth.GoogleAuthenticator -> D:\myproject\Presentation\Nop.Web\Plugins\MultiFactorAuth.GoogleAuthenticator\Nop.Plugin.MultiFactorAuth.GoogleAuthenticator.dll
  Nop.Plugin.Payments.PayPalStandard -> D:\myproject\Presentation\Nop.Web\Plugins\Payments.PayPalStandard\Nop.Plugin.Payments.PayPalStandard.dll
  Nop.Plugin.DiscountRules.CustomerRoles -> D:\myproject\Presentation\Nop.Web\Plugins\DiscountRules.CustomerRoles\Nop.Plugin.DiscountRules.CustomerRoles.dll
  Nop.Plugin.Payments.Manual -> D:\myproject\Presentation\Nop.Web\Plugins\Payments.Manual\Nop.Plugin.Payments.Manual.dll
  Nop.Plugin.Widgets.NivoSlider -> D:\myproject\Presentation\Nop.Web\Plugins\Widgets.NivoSlider\Nop.Plugin.Widgets.NivoSlider.dll
  Nop.Plugin.Misc.WebApi.Frontend -> D:\myproject\Presentation\Nop.Web\Plugins\Misc.WebApi.Frontend\Nop.Plugin.Misc.WebApi.Frontend.dll
  Nop.Plugin.Widgets.AccessiBe -> D:\myproject\myapp\
  Nop.Plugin.Widgets.GoogleAnalytics -> D:\myproject\Presentation\Nop.Web\Plugins\Widgets.GoogleAnalytics\Nop.Plugin.Widgets.GoogleAnalytics.dll
  Innovapps.Nop.Plugin.Misc.BusinessCentralErp -> D:\myproject\Presentation\Nop.Web\Plugins\Misc.BusinessCentralErp\Innovapps.Nop.Plugin.Misc.BusinessCentralErp.dll
  Nop.Plugin.Tax.FixedOrByCountryStateZip -> D:\myproject\myapp\
  Nop.Plugin.Pickup.PickupInStore -> D:\myproject\myapp\
  Nop.Plugin.ExternalAuth.Facebook -> D:\myproject\Presentation\Nop.Web\Plugins\ExternalAuth.Facebook\Nop.Plugin.ExternalAuth.Facebook.dll
  Nop.Plugin.Shipping.FixedByWeightByTotal -> D:\myproject\myapp\
  Nop.Plugin.Shipping.UPS -> D:\myproject\Presentation\Nop.Web\Plugins\Shipping.UPS\Nop.Plugin.Shipping.UPS.dll
  Nop.Plugin.Payments.Manual -> D:\myproject\myapp\
  Nop.Plugin.Widgets.GoogleAnalytics -> D:\myproject\myapp\
  Nop.Plugin.Payments.PayPalStandard -> D:\myproject\myapp\
  Nop.Plugin.Widgets.NivoSlider -> D:\myproject\myapp\
  Nop.Plugin.Misc.Sendinblue -> D:\myproject\Presentation\Nop.Web\Plugins\Misc.Sendinblue\Nop.Plugin.Misc.Sendinblue.dll
  Nop.Plugin.Misc.WebApi.Frontend -> D:\myproject\myapp\
  Nop.Plugin.Payments.CheckMoneyOrder -> D:\myproject\myapp\
  Nop.Plugin.DiscountRules.CustomerRoles -> D:\myproject\myapp\
  Nop.Plugin.Shipping.UPS -> D:\myproject\myapp\
  Nop.Plugin.MultiFactorAuth.GoogleAuthenticator -> D:\myproject\myapp\
  Nop.Plugin.ExternalAuth.Facebook -> D:\myproject\myapp\
relative path: D:\myproject\Presentation\Nop.Web\Views\_ViewImports.cshtml, D:\myproject\Plugins\Nop.Plugin.Widgets.FacebookPixel\Views\_ViewImports.cshtml. [D:\myproject\Plugins\Nop.Plugin.Widgets.FacebookPixel\Nop.Plugin.Widgets.FacebookPixel.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ConflictResolution.targets(112,5): error NETSDK1152: Found multiple publish output files with the same relative path: D:\myproject\Presentation\Nop.Web\Views\_ViewImports.cshtml, D:\myproject\Plugins\Nop.Plugin.Shipping.ShipStation\Views\_ViewImports.cshtml. [D:\myproject\Plugins\Nop.Plugin.Shipping.ShipStation\Nop.Plugin.Shipping.ShipStation.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ConflictResolution.targets(112,5): error NETSDK1152: Found multiple publish output files with the same relative path: D:\myproject\Presentation\Nop.Web\Views\_ViewImports.cshtml, D:\myproject\Plugins\Nop.Plugin.Widgets.What3words\Views\_ViewImports.cshtml. [D:\myproject\Plugins\Nop.Plugin.Widgets.What3words\Nop.Plugin.Widgets.What3words.csproj]
  Nop.Tests -> D:\myproject\myapp\
C:\Program Files\dotnet\sdk\6.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ConflictResolution.targets(112,5): error NETSDK1152: Found multiple publish output files with the same relative path: D:\myproject\Presentation\Nop.Web\Views\_ViewImports.cshtml, D:\myproject\Plugins\Nop.Plugin.Payments.PayPalCommerce\Views\_ViewImports.cshtml. [D:\myproject\Plugins\Nop.Plugin.Payments.PayPalCommerce\Nop.Plugin.Payments.PayPalCommerce.csproj]C:\Program Files\dotnet\sdk\6.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ConflictResolution.targets(112,5): error NETSDK1152: Found multiple publish output files with the same relative path: D:\myproject\Presentation\Nop.Web\Views\_ViewImports.cshtml, D:\myproject\Plugins\Nop.Plugin.Tax.Avalara\Views\_ViewImports.cshtml. [D:\myproject\Plugins\Nop.Plugin.Tax.Avalara\Nop.Plugin.Tax.Avalara.csproj]C:\Program Files\dotnet\sdk\6.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ConflictResolution.targets(112,5): error NETSDK1152: Found multiple publish output files with the same relative path: D:\myproject\Presentation\Nop.Web\Views\_ViewImports.cshtml, D:\myproject\Plugins\Nop.Plugin.Shipping.EasyPost\Views\_ViewImports.cshtml. [D:\myproject\Plugins\Nop.Plugin.Shipping.EasyPost\Nop.Plugin.Shipping.EasyPost.csproj]
```

**Comment by skoshelev** [comment]
Hi, @Manuel-Innovapps! Seems like a specific CI issue as we prepare our packages using publish and do publish in the same way when building a docker container and didn't encounter this issue, please try the tips on [this link](https://stackoverflow.com/questions/69919664/publish-error-found-multiple-publish-output-files-with-the-same-relative-path)

**Comment by Manuel-Innovapps** [comment]
Hey @skoshelev 

That does not only seem to be a bug on my machine or on my ci. 
I just forked this repo, to show the problem:

URL: https://github.com/Manuel-Innovapps/nopCommerce/
CI Action: https://github.com/Manuel-Innovapps/nopCommerce/blob/develop/.github/workflows/dotnet.yml
Failing Pipeline Run: https://github.com/Manuel-Innovapps/nopCommerce/runs/4936104276?check_suite_focus=true

I also tested locally to add 
```
<ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles>
```
as suggested on SO, but that does not to seem to solve the error.

Here is the change in the repo:
Commit: https://github.com/Manuel-Innovapps/nopCommerce/commit/d44a831db601e6bdd9d3d37f213701941e650697
Failing Workflow Run: https://github.com/Manuel-Innovapps/nopCommerce/runs/4936174465?check_suite_focus=true

Hope this helps you diagnosing the error.

**Comment by Manuel-Innovapps** [comment]
I think I just found a workaround.

I was trying to do 
```
dotnet publish
```
but this brings the problem above.

Instead doing
```
dotnet publish ./Presentation/Nop.Web/Nop.Web.csproj
```
seems to be working just fine.


I personally would still consider this a workaround and not a solution for the issue, as `dotnet publish` seems to be an official dotnet command that should also work for nopCommerce.

**Comment by Manuel-Innovapps** [comment]
Maybe a small explanation why this situation happend at all:
Our customer set up an Azure Tenant and invited us to set up some ci/cd.
We went into the deployment center in Azure, selected GitHub and GitHub Actions as source, and configured the repo etc. Then Azure created an GitHub actions workflow for us, containing the script shown above.

This was what led me to believe that this script was correct, and is also the reason I opened this ticket.

**Comment by skoshelev** [comment]
Hi @Manuel-Innovapps. I analyzed the situation again. Of course the dotnet publish command is valid, but not within nopCommerce. Our project consists of many libraries, and when running a common command, dotnet tries to build and publish all the projects one by one, which, in principle, does not correspond to the or task. In addition, we have specific publishing settings specified in the Nop.Web.csproj file. Therefore, it is necessary to publish only the Nop.Web project, and it will already pull up the necessary assemblies on its own.

Also see this part of our [documentation](https://docs.nopcommerce.com/developer/tutorials/azure-publish.html#publish-project-nopweb). It also indicates that it is necessary to publish the Nop.Web assembly and not the entire solution

Closed #6096
